core/templates/services/playthrough.service.ts

```diff
@@ -27,373 +26,252 @@ import { LearnerActionObjectFactory } from
-            // In cases where maximum number of playthroughs already exists, the
-            // above flag is not True and playthrough ID is not set.
-            PlaythroughService.playthrough.playthroughId = (
-              response.playthrough_id);
-          }
-        });
-      }
-    }
+class EarlyQuitTracker {
+  constructor(private timeSpentInStateSecs: number = null) {}
 
-    private getFullPlaythroughUrl(): string {
-      return this.urlInterpolationService.interpolateUrl(
-        ServicesConstants.STORE_PLAYTHROUGH_URL, {
-          exploration_id: PlaythroughService.playthrough.expId
-        });
-    }
+  isIssue(): boolean {
+    return (
+      this.timeSpentInStateSecs !== null &&
+      this.timeSpentInStateSecs < ServicesConstants.EARLY_QUIT_THRESHOLD_IN_SECS
+    );
+  }
 
-    private isPlaythroughDiscarded(): boolean {
-      return !this.explorationFeaturesService.isPlaythroughRecordingEnabled() ||
-        !PlaythroughService.isLearnerInSamplePopulation;
-    }
+  recordExplorationQuit(timeSpentInStateSecs: number): void {
+    this.timeSpentInStateSecs = timeSpentInStateSecs;
+  }
+}
 
-    initSession(
-        explorationId: string, explorationVersion: number,
-        playthroughProbability: number): void {
-      PlaythroughService.isLearnerInSamplePopulation =
-        this.determineIfLearnerIsInSamplePopulation(playthroughProbability);
-      PlaythroughService.playthrough = this.playthroughObjectFactory.createNew(
-        null, explorationId, explorationVersion, null, {}, []);
-      PlaythroughService.expStopwatch = this.stopwatchObjectFactory.create();
-      PlaythroughService.multipleIncorrectStateName = {
-        state_name: null,
-        num_times_incorrect: null
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughService {
+  private explorationId?: string = null;
+  private explorationVersion?: number = null;
+  private isLearnerInSamplePopulation?: boolean = null;
+
+  private eqTracker?: EarlyQuitTracker = null;
+  private cstTracker?: CyclicStateTransitionsTracker = null;
+  private miaTracker?: MultipleIncorrectAnswersTracker = null;
+  private playthrough?: Playthrough = null;
+  private stopwatch?: Stopwatch = null;
+  private expDurationInSecs?: number = null;
+
+  constructor(
+      private explorationFeaturesService: ExplorationFeaturesService,
+      private learnerActionObjectFactory: LearnerActionObjectFactory,
+      private playthroughBackendApiService: PlaythroughBackendApiService,
+      private playthroughObjectFactory: PlaythroughObjectFactory,
+      private stopwatchObjectFactory: StopwatchObjectFactory) {}
+
+  /**
+   * The ordering of checks in this method prioritizes the following types of
+   * playthroughs:
+   *    1. MultipleIncorrectSubmissionsIssue
+   *    2. CyclicStateTransitionsIssue
+   *    3. EarlyQuitIssue
+   */
+  private classifyPlaythrough(): void {
+    if (this.miaTracker !== null && this.miaTracker.isIssue()) {
```

Comment:
mia --> mis, here and below? (Nomenclature does not match your comment/docstring above.)

Reply:
Done.